src/platform/Zephyr/SystemTimeSupport.cpp

```diff
@@ -35,32 +35,58 @@ namespace System {
 namespace Clock {
 
 namespace Internal {
+
 ClockImpl gClockImpl;
+
 } // namespace Internal
 
-Microseconds64 ClockImpl::GetMonotonicMicroseconds64(void)
+namespace {
+
+// Last known UTC time in Unix format.
+#ifdef CHIP_DEVICE_CONFIG_FIRMWARE_BUILD_UNIX_TIME
+Clock::Microseconds64 gRealTime = Microseconds64(CHIP_DEVICE_CONFIG_FIRMWARE_BUILD_UNIX_TIME * UINT64_C(1000000));
```

Comment:
I suspect you could make the RHS be `Seconds64(CHIP_DEVICE_CONFIG_FIRMWARE_BUILD_UNIX_TIME)` and this would either work, or worst case work with a static_cast<> to `Microseconds64`, without needing the manual multiplication.

Also, why the `Clock::` prefixes thoughout here?  This is all inside `namespace Clock`.